packages/next/src/client/components/http-error-fallback-boundary.tsx

```diff
@@ -62,45 +80,46 @@ class NotFoundErrorBoundary extends React.Component<
   static getDerivedStateFromError(error: any) {
     if (isNotFoundError(error)) {
       return {
-        notFoundTriggered: true,
+        triggeredStatus: HTTPErrorStatus.NOT_FOUND,
       }
     }
     // Re-throw if error is not for 404
     throw error
   }
 
   static getDerivedStateFromProps(
-    props: NotFoundErrorBoundaryProps,
-    state: NotFoundErrorBoundaryState
-  ): NotFoundErrorBoundaryState | null {
+    props: HTTPErrorFallbackErrorBoundaryProps,
+    state: HTTPErrorBoundaryState
+  ): HTTPErrorBoundaryState | null {
     /**
      * Handles reset of the error boundary when a navigation happens.
      * Ensures the error boundary does not stay enabled when navigating to a new page.
      * Approach of setState in render is safe as it checks the previous pathname and then overrides
      * it as outlined in https://react.dev/reference/react/useState#storing-information-from-previous-renders
      */
-    if (props.pathname !== state.previousPathname && state.notFoundTriggered) {
+    if (props.pathname !== state.previousPathname && state.triggeredStatus) {
       return {
-        notFoundTriggered: false,
+        triggeredStatus: undefined,
         previousPathname: props.pathname,
       }
     }
     return {
-      notFoundTriggered: state.notFoundTriggered,
+      triggeredStatus: state.triggeredStatus,
       previousPathname: props.pathname,
     }
   }
 
   render() {
-    if (this.state.notFoundTriggered) {
+    if (this.state.triggeredStatus === HTTPErrorStatus.NOT_FOUND) {
+      const [fallback, styles] = this.props.notFound || []
```

Comment:
why are styles external of the the not found component? shouldn't we just define the notFound node as a fragment and pass it in with styles included?

Reply:
I agree. We already handle it like that [here](https://github.com/vercel/next.js/pull/72586/files#diff-4de93b8094b02442001721bb5c9164fdcd9127a37276e5d3897dd34cb2a13342R629-R632) and [here](https://github.com/vercel/next.js/pull/72586/files#diff-4de93b8094b02442001721bb5c9164fdcd9127a37276e5d3897dd34cb2a13342R697-R700), so might as well pass in a single node and avoid the tuple prop.

Reply:
single node sounds good, I only refactored `notFound` for now as it's relevant and we could refactor other props like `template` etc later